packages/button/src/LionButton.js

```diff
@@ -167,31 +171,13 @@ export class LionButton extends DisabledWithTabIndexMixin(SlotMixin(LitElement))
     }
   }
 
-  _redispatchClickEvent(oldEvent) {
-    // replacing `MouseEvent` with `oldEvent.constructor` breaks IE
-    const newEvent = new MouseEvent(oldEvent.type, oldEvent);
-    newEvent.__isRedispatchedOnNativeButton = true;
-    this.__enforceHostEventTarget(newEvent);
-    this._nativeButtonNode.dispatchEvent(newEvent);
-  }
-
   /**
-   * Prevent normal click and redispatch click on the native button unless already redispatched.
+   * Prevent normal click and redispatch submit on the native button unless already redispatched.
    */
-  __clickDelegationHandler(e) {
-    if (!e.__isRedispatchedOnNativeButton) {
-      e.stopImmediatePropagation();
-      this._redispatchClickEvent(e);
-    }
-  }
-
-  __enforceHostEventTarget(event) {
-    try {
-      // this is for IE11 (and works in others), because `Object.defineProperty` does not give any effect there
-      event.__defineGetter__('target', () => this); // eslint-disable-line no-restricted-properties
-    } catch (error) {
-      // in case `__defineGetter__` is removed from the platform
-      Object.defineProperty(event, 'target', { writable: false, value: this });
+  __clickDelegationHandler() {
+    if (this.type === 'submit' && this._nativeButtonNode.form) {
+      this._nativeButtonNode.form.dispatchEvent(new Event('submit'));
+      this._nativeButtonNode.form.submit();
```

Comment:
Did we need both in the end?

Reply:
Yes ;) submit() wouldn't trigger @submit handler, and dispatchEvent wouldn't cause the actual form to submit and cause page refresh, so both are necessary.